pkg/kopia/cli/internal/flag/flag.go

```diff
@@ -0,0 +1,81 @@
+// Copyright 2024 The Kanister Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package flag
+
+import (
+	"github.com/kanisterio/safecli"
+)
+
+// Applier applies flags/args to the command.
+type Applier interface {
+	// Apply applies the flags/args to the command.
+	Apply(cli safecli.CommandAppender) error
+}
+
+// Apply appends multiple flags to the CLI.
+// If any of the flags encounter an error during the Apply process,
+// the error is returned and no changes are made to the CLI.
+// If no error, the flags are appended to the CLI.
+func Apply(cli safecli.CommandAppender, flags ...Applier) error {
+	// create a new sub builder which will be used to apply the flags
+	// to avoid mutating the CLI if an error is encountered.
+	sub := safecli.NewBuilder()
+	for _, flag := range flags {
+		if flag == nil { // if the flag is nil, skip it
+			continue
+		}
+		if err := flag.Apply(cli); err != nil {
```

Comment:
Is that supposed to be flag.Apply(sub)? Otherwise what's the use of sub?

Reply:
Good catch! Fixed.